### DISCUSSION TITLE
Is it possible to bake Teleport into a Packer-based golden image configuration?

### DISCUSSION THREAD

**Discussion by vincentracine** [gravitational/teleport#13872] [Q&A] [answered]
We are generating a golden disk image using Packer (https://www.packer.io/). The disk image is subsequently flashed onto a Raspberry Pi that is embedded within our edge-device product. Is there a configuration available that would allow us to achieve this use case? From what I can read, I think that the AWS IAM setup would best suited for us?

**Comment by webvictim** [comment, marked as answer]
The AWS IAM setup will only work on machines running inside AWS which are allowed to query the identity of the account that owns them. It sounds like you're shipping machines that would run outside AWS?

You might like to look into [static tokens](https://goteleport.com/docs/setup/admin/adding-nodes/#an-insecure-alternative-static-tokens) which will allow you to set a token up ahead of time allowing nodes to join your cluster. That way you can install Teleport inside the golden image, configure it to run on startup and join using the predefined static token.

If you know your machines will be coming online within a fixed timeframe, it's better from a security perspective to use a [dynamic token](https://goteleport.com/docs/setup/admin/adding-nodes/#generate-a-token) with a TTL configured instead. You could add a process which adds a new token with a fixed TTL every time the image is rebuilt, rather than having one permanently-valid token.

I should point out as well that if you're already baking AWS credentials into the image for other reasons, you could theoretically provide them to Teleport's environment and use the IAM join method instead.

**Reply by vincentracine** [reply]
Understood, the AWS IAM setup is not appropriate for this use case. 

Static tokens seem like the best way to go for now. Alternatively, as you mention, we could generate a token with a semi-long TTL, such as 3 months, and know not to flash images that have "expired". 

I appreciate the quick response time!!
